.travis.yml

```diff
@@ -15,9 +15,10 @@ jobs:
   include:
     - if: type = pull_request
       name: checkpatch
-      script:
+      script: # FIXME: replace with TRAVIS_COMMIT_RANGE or similar
         - SHA_PR=`git log --oneline -1 | sed -rn "s/.*Merge (.*) into.*/\1/p"`
         - SHA_MAIN=`git log --oneline -1 | sed -rn "s/.*Merge .* into (.*)/\1/p"`
+        - git fetch https://github.com/marc-hb/linux checkpatch-strict-not-fatal && git cherry-pick FETCH_HEAD
```

Comment:
why not add an SOF-specific script in the SOF tree, and avoid dependencies something we don't see and can't make PRs on.

Reply:
As I mentioned in the first github comment, this location wasn't meant to be final.

I can add this  two lines long checkpatch commit as a .patch in the `sof-dev` branch and then git apply from this same .travis.yml location.

Earlier I suggested a dedicated, thesofproject/linux/ branch different from `sof-dev` to store these two lines because https://thesofproject.github.io/latest/contribute/linux/development_tree.html#upstream-merge-rebase says:
> After the two separate operations of merge and rebase on the two branches, these two branches (sof-dev and sof-dev-rebase) should again be identical.

I don't think you want me to edit `scripts/checkpatch.pl` in sof-dev directly because
1. "these two branches (sof-dev and sof-dev-rebase) should again be identical."
2. More importantly, this wouldn't be swapping Travis and Jenkins anymore, this would make strict-only failures green _everywhere_.

Let me know your preference.

Reply:
What I meant was *ADDING* a new scripts/sof-checkpatch-strict.pl, and we update this file on each kernel revision to align with checkpatch.pl

Reply:
OK, no problem I can add `scripts/sof-checkpatch-strict-not-fatal.pl`. After reading https://thesofproject.github.io/latest/contribute/linux/development_tree.html#upstream-merge-rebase that seemed like more maintenance than applying a 2-lines long patch on the fly but you are maintaining these branches and I'm not :-)

Reply:
yes, but that process was for developers. This is an infrastructure patch so we can be flexible. Same as for the .yaml thing

Reply:
> This is an infrastructure patch so we can be flexible. 

If a somewhat "impure" sof-dev branch is acceptable, then I have a better checkpatch option than copying the entire script. See https://github.com/thesofproject/linux/compare/topic/sof-dev...marc-hb:checkpatch-no-strict-fails

Demo:
```
./scripts/checkpatch.pl -g b943f7980113                            # $? = 0
./scripts/checkpatch.pl -g b943f7980113 --strict                   # $? = 1
./scripts/checkpatch.pl -g b943f7980113 --strict --no-strict-fails # $? = 0
```
It is a bit more longer than the 2 lines and just 1 hunk of https://github.com/thesofproject/linux/compare/topic/sof-dev...marc-hb:checkpatch-strict-not-fatal so the risk of git conflict is a bit higher but barely: 5 lines and 3 hunks. Smaller than I expected.

Reply:
I honestly prefer a different file to avoid merge conflicts which may or may not work and result in CI flagging false positive or not flagging true issues. We'd know it's our own thing.

Reply:
@plbossart Hmm, but if we anyways need to take a look at checkpatch changes at every rebase, why not maintain a patch that modifies checkpatch.pl directly in sof-dev? Anyways us maintainers need to take a look at the file at each rebase, so why not let git help us. When we upload a new rebase PR, we'll get results whether we broke the CI checks not (i.e. whether we screwed up the checkpatch rebase or not).

Reply:
Of course an even simpler solution that doesn't require any checkpatch change at all is to simply drop `--strict` in `.travis.yml`. I mean is there really a need to duplicate the display of `--strict` failures in Travis (hidden behind a green mark that most people will never open...) once those same `--strict` failures will be visible again on the PR main page under a red line named "checkpatch"?

PS: pushing code first and then discussing "requirements"  later deep down this PR is not ideal but was intentional. I could have opened a github issue to discuss requirements first but: it's generally hard to get code reviews, it's harder to get CI code reviews and IMHO it's nearly impossible to get reviews of micro CI requirements like this one. Why I opted for the  "send patches" + demo approach.

Reply:
we should really revisit the whole thing. IIRC Travis has different - stricter - compilation options than Jenkins.